pkg/serializer/internal/metrics/sketch_series_list.go

```diff
@@ -320,6 +325,9 @@ func (sl SketchSeriesList) SplitPayload(times int) ([]marshaler.AbstractMarshale
 		ss := sl.Current()
 		sketches = append(sketches, ss)
 	}
+	if len(sketches) == 0 {
+		return nil, fmt.Errorf("Not possible to split payload since all sketches have been read already")
```

Comment:
If we get here because a previous method has consumed all payloads already, it doesn't seem like an error condition to me. Would it make sense to return an empty slice (or nil) and no error in this case?